src/virtualenv/run/plugin/discovery.py

```diff
@@ -13,10 +13,13 @@ def get_discover(parser, args):
         title="discovery",
         description="discover and provide a target interpreter",
     )
+    choices = _get_default_discovery(discover_types)
+    # prefer the builtin if present, otherwise fallback to first defined type
+    choices = sorted(choices, key=lambda a: 0 if a == "builtin" else 1)
```

Comment:
This these lines have been borrowed from here: https://github.com/pypa/virtualenv/blob/0262fa6d9a0f8b45c80e6b2f82109ab1250ec336/src/virtualenv/run/plugin/creators.py#L52-L53